daisy_workflows/image_test/network/network-tester.py

```diff
@@ -39,28 +43,51 @@ def GetHostname(addr, timeout=10):
   return urllib2.urlopen(URL, None, timeout).read()
 
 
-def TestIPAlias(testee, ip_alias, ip_mask):
-  testee_hostname = GetHostname(testee)
+def TestIPAlias(instance, ip_alias, ip_mask):
+  testee_hostname = GetHostname(instance)
+  cidr = "%s/%s" % (ip_alias, ip_mask)
 
   # caching ips out of range for using it on negative tests
   ips = list(genips.GenIPs(ip_alias, ip_mask))
 
   # positive test on expected IPs
   for ip in ips:
-    if testee_hostname != GetHostname(ip):
-      raise Exception("Alias hostname should be the same from host machine.")
+    try:
+      if testee_hostname != GetHostname(ip):
+        raise Exception("Alias hostname should be the same from host machine.")
+    except urllib2.URLError:
+      raise RequestError("ip_alias", instance, cidr)
 
   # negative test
   superset_ips = list(genips.GenIPs(ip_alias, int(ip_mask) - 1))
   invalid_ip = superset_ips[superset_ips.index(ips[0]) - 1]
   try:
     if testee_hostname == GetHostname(invalid_ip):
-      e = "IP alias out of range (%s) should not respond." % invalid_ip
-      raise Exception(e)
-  except (RequestError, urllib2.URLError):
+      raise RequestError("ip_alias", instance, cidr, positive_test=False)
+  except urllib2.URLError:
+    # great, it didn't even respond! So no machine is using that ip.
     pass
 
 
+def SetIPAlias(MD, instance, ip_alias=None, ip_mask=None):
+  # by default, remove the aliasing
+  alias_ip_ranges = []
+  if ip_alias:
+    cidr = "%s/%s" % (ip_alias, ip_mask)
+    alias_ip_ranges = [{'ipCidrRange': cidr}]
+
+  info = {
+      'aliasIpRanges': alias_ip_ranges,
+      # a previous fingerprint is needed in order to change the instance
+      'fingerprint': MD.GetInstanceIfaces(instance)[0][u'fingerprint']
+  }
+  MD.Wait(MD.SetInstanceIface(instance, info))
+
+  # The operation is completed but it might not guarantee that the agent has
+  # already applied the rule. For that reason, wait
+  time.sleep(30)
```

Comment:
I didn't like this but I didn't come up with a better method. Anyone?

Reply:
We could watch the serial port, but thats kinda complicated, this sleep is more than enough. If we wanted to clean it up we can later,